### PR TITLE
Add release instructions for static analysis and 3rd party deps (CXX-3009, CXX-3023)

### PR DESCRIPTION
**Pull request by eramongodb**
Resolves CXX-3009 and CXX-3023.

This PR satisfies the "Static Analysis Report" and "3rd Party Dependency Vulnerabilities" report requirements for SSDLC.

3rd Party Dependencies (3PDs) are documented by `etc/purls.txt` and the corresponding `etc/cyclonedx.sbom.json` document, including their version and license. Any vulnerabilities which affect the "standard release product" for the MongoDB C++ Driver are reported in `etc/third_party_vulnerabilities.md`, which will be bundled with the signed release tarball alongside the final SSDLC Compliance Report per release. See `etc/third_party_vulnerabilities.md` for the definition of the "standard release product".

Static analysis issues are tracked and documented by a Google Spreadsheet which will be cloned and linked-to in the final SSDLC Compliance Report per release. This is expected to be a temporary solution until better automation, tooling, and integrations are implemented. Details to follow in the subsequent PR which will introduce `etc/ssdlc_compliance_report.md`.

To better track the status of 3PD vulnerabilities (currently, none), a new Evergreen task tests for significant changes to the augmented SBOM document. ~The `etc/cyclonedx.sbom.json` document serves as both the "SBOM Lite" and "augmented SBOM" document, as no observable difference between the two types of files could be found thus far. Should there be significant differences,~ an `etc/augmented.sbom.json` file is expected to be committed alongside `etc/cyclonedx.sbom.json` in the future. (Update: confirmed, the Augmented SBOM should not be used as the SBOM Lite.)

### CODE REVIEW

File: etc/releasing.md [hunk 1]
```diff
@@ -23,25 +23,51 @@ by changes in the new release.
...
+      "SILK_CLIENT_ID=${SILK_CLIENT_ID:?}" \
+      "SILK_CLIENT_SECRET=${SILK_CLIENT_SECRET:?}"
+  ) \
+  -it --rm -v "$(pwd):/pwd" \
+  artifactory.corp.mongodb.com/release-tools-container-registry-public-local/silkbomb:1.0 \
+  update -p "/pwd/etc/purls.txt" -i "/pwd/etc/cyclonedx.sbom.json" -o "/pwd/etc/cyclonedx.sbom.json"
 ```
 
-## Check Snyk
+Ensure any vulnerabilities with severity "High" or greater have a corresponding JIRA ticket (CXX or VULN) that is scheduled to be resolved within its remediation timeline.
```
Comment (kevinAlbs) [inline etc/releasing.md:58]:
```suggestion
Ensure any vulnerabilities with severity "Medium" or greater have a corresponding JIRA ticket (CXX or VULN) that is scheduled to be resolved within its [remediation timeline](https://docs.google.com/document/d/1u0m4Kj2Ny30zU74KoEFCN4L6D_FbEYCaJ3CQdCYXTMc/edit#bookmark=id.l09k96qt24jm).
```

I expect "Medium" is needed from [MongoDB Software Security Development Lifecycle Policy  > Third party dependencies](https://docs.google.com/document/d/1u0m4Kj2Ny30zU74KoEFCN4L6D_FbEYCaJ3CQdCYXTMc/edit#bookmark=id.l09k96qt24jm)

File: etc/third_party_vulnerabilities.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# 3rd Party Dependency Vulnerabilities
+
+This document tracks vulnerabilities in 3rd party dependencies that directly affect the MongoDB C++ Driver standard release product.
+
+> [!IMPORTANT]
+> The "standard release product" is defined as the set of files which are _installed_ by a configuration, build, and install of the MongoDB C++ Driver. This includes static/shared library files, header files, and packaging files for supported build configurations. Vulnerabilities for 3rd party dependencies that are bundled with or required by the standard release product are reported in this document.
```
Comment (kevinAlbs) [inline etc/third_party_vulnerabilities.md:6]:
```suggestion
> The "standard release product" is defined as the set of files which are _installed_ by a configuration, build, and install of the MongoDB C++ Driver. This includes static/shared library files, header files, and packaging files for supported build configurations. Vulnerabilities for 3rd party dependencies that are bundled with the standard release product are reported in this document.
```

Suggest removing "or required by". [Scope: SSDLC Policy Conformance for Client Libs](https://docs.google.com/document/d/1mXxDdY38nGVKWJhIbH83X6mcHsxllAiunH2hg-NdDsY/edit) notes:

> For purposes of reporting and vulnerability tracking, third-party dependencies only refers to bundled dependencies that ship with a driver.

Example: I expect boost is excluded due to not being bundled. Yet I expect boost could be considered a required dependency (it is one option to satisfy the required polyfill dependency).

**Review by kevinAlbs — APPROVED** [review]
LGTM with minor comments addressed.

I liked the proposed `silk-check-augmented-sbom` and committing the Augmented SBOM. I expect the task will give advanced notice to address a 3PD vulnerability before a release. I had incorrectly assumed the Augmented SBOM was not intended to be committed.

**Comment by eramongodb** [comment]
Re-requesting reviews due to some notable changes. Latest changes verified by [this patch](https://spruce.mongodb.com/version/666201f9aace5e0007fdb35e).

- `etc/augmented.json.sbom` committed into repository as the Augmented SBOM.
- Better care is taken to distinguish the SBOM Lite document from the Augmented SBOM document, as they are not equivalent: the latter contains up-to-date vulnerabilities, the former does not; including them in the former will make the vulnerabilities "sticky" when generating the Augmented SBOM).
- Added commands to ensure a latest version of SilkBomb is used in update/download commands.
- Improved the Augmented SBOM diff to utilize SilkBomb flags, omitting need to filter `.metadata`.

File: etc/augmented.sbom.json
[the remark this answers is not in the document]
Comment (eramongodb) [reply etc/augmented.sbom.json:1]: It only needs to be available as an artifact for a given release, and it doesn't need to be regularly committed either. Therefore, the proposed release instructions state that this file only needs to be updated if there are significant changes in the latest Augmented SBOM relative to what is currently committed (as will be tracked by the EVG task). In effect, its presence in the repo isn't just for reporting purposes, but also to enable the tracking of updates to the Augmented SBOM via the EVG task.

**Review by vector-of-bool — APPROVED** [review]
LGTM

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**